x-pack/elastic-agent/CHANGELOG.next.asciidoc

```diff
@@ -39,6 +39,7 @@
 - Increase checkin grace period to 30 seconds {pull}23568[23568]
 - Fix libbeat from reporting back degraded on config update {pull}23537[23537]
 - Fix issues with dynamic inputs and conditions {pull}23886[23886]
+- Fix bad substitution of API key. {pull}24036[24036]
```

Comment:
Is this required? I assume the fix goes into the same version (`7.12`) as the changes that broke the setup.

Reply:
I don't have anything  strong opinion about it. This is a change (bugfix) hence the CHANGELOG entry. I will backport it to `7.x` once this is merged.

Reply:
Maybe I am not up-to-date with how this works in the beats repo, but since Changelog entries are usually user facing, I am used to only include bugfixes that are user facing (which is not the case if introduced & fixed in the same version). Up to you though whether or not to keep it.